packages/core-components/src/layout/SignInPage/GuestUserIdentity.ts

```diff
@@ -19,14 +19,44 @@ import {
   ProfileInfo,
   BackstageUserIdentity,
 } from '@backstage/core-plugin-api';
+import { Config } from '@backstage/config';
+import nJwt from 'njwt';
+import { GUEST_USER_PRIVATE_KEY } from '@backstage/plugin-auth-common';
 
 export class GuestUserIdentity implements IdentityApi {
+  private idToken?: string;
+
+  static fromConfig(config?: Config): GuestUserIdentity {
+    const newIdentity = new GuestUserIdentity();
+    if (config?.getOptionalBoolean(`auth.allowGuestMode`)) {
+      newIdentity.setIdToken(newIdentity.issueToken());
+    }
+    return newIdentity;
+  }
+
+  private setIdToken(token: string) {
+    this.idToken = token;
+  }
+
+  private issueToken(): string {
+    const userEntityRef = `user:default/guest`;
```

Comment:
@Rugvip Should we formalize a "unique enough" actual ref for this, or should we make a completely new token type instead? Possibly by just leaving out `sub` if possible? If we keep a ref shaped sub, it should probably be something longer and weirder, like `user:backstage-auth-backend-generated-identities/guest` or whatnot, in a dedicated const

Reply:
Separately: it is kinda neat if you use `stringifyEntityRef` even though it seems wasteful, because it encapsulates some of the intricacies of how these are formed. Your string is correct as it stands indeed, nothing wrong as such, but over time it proves helpful and informative.

```ts
stringifyEntityRef({
  kind: 'User',
  namespace: 'default',
  name: 'guest',
});
```

Reply:
I am all for setting standards on user entity creation. I went ahead and changed it to use `stringifyEntityRef`